src/all/deviantart/src/eu/kanade/tachiyomi/extension/all/deviantart/DeviantArt.kt

```diff
@@ -0,0 +1,186 @@
+        }
+        manga.setUrlWithoutDomain(response.request.url.toString())
+        return manga
+    }
+
+    override fun chapterListRequest(manga: SManga): Request {
+        val pathSegments = getMangaUrl(manga).toHttpUrl().pathSegments
+        val username = pathSegments[0]
+        val folderId = pathSegments[2]
+
+        val url = backendBuilder()
+            .addPathSegment("rss.xml")
+            .addQueryParameter("q", "gallery:$username/$folderId")
+            .build()
+
+        return GET(url, headers)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoupXml()
+        val chapterList = parseToChapterList(document).toMutableList()
+        var nextUrl = document.selectFirst("[rel=next]")?.attr("href")
+
+        while (nextUrl != null) {
+            val newRequest = GET(nextUrl, headers)
+            val newResponse = client.newCall(newRequest).execute()
+            val newDocument = newResponse.asJsoupXml()
+            val newChapterList = parseToChapterList(newDocument)
+            chapterList.addAll(newChapterList)
+
+            nextUrl = newDocument.selectFirst("[rel=next]")?.attr("href")
+        }
+
+        return indexChapterList(chapterList.toList())
+    }
+
+    private fun parseToChapterList(document: Document): List<SChapter> {
+        val items = document.select("item")
+        return items.map {
+            val chapter = SChapter.create()
+            chapter.setUrlWithoutDomain(it.selectFirst("link")!!.text())
+            chapter.apply {
+                name = it.selectFirst("title")!!.text()
+                date_upload = parseDate(it.selectFirst("pubDate")!!.text())
+                scanlator = it.selectFirst("media|credit")!!.text()
+            }
+        }
+    }
+
+    private fun indexChapterList(chapterList: List<SChapter>): List<SChapter> {
+        // DeviantArt allows users to arrange galleries arbitrarily so we will
+        // primitively index the list by checking the first and last dates
+        return if (chapterList.first().date_upload > chapterList.last().date_upload) {
+            chapterList.mapIndexed { i, chapter ->
+                chapter.apply { chapter_number = chapterList.size - i.toFloat() }
+            }
+        } else {
+            chapterList.mapIndexed { i, chapter ->
+                chapter.apply { chapter_number = i.toFloat() + 1 }
+            }
+        }
+    }
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        val url = backendBuilder()
+            .addPathSegment("oembed")
+            .addQueryParameter("url", baseUrl + chapter.url)
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val imageUrl = Json.parseToJsonElement(response.body.string())
+            .jsonObject["url"]!!
+            .jsonPrimitive
+            .content
```

Comment:
Usually I would say you should use DTOs, but since it's only used here I guess it's ok.